lib/exercises.js

```diff
@@ -1,29 +1,74 @@
 // This method will return an array of arrays.
 // Each subarray will have strings which are anagrams of each other
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: o(n)
+// Space Complexity: o(n)
 function grouped_anagrams(strings) {
-  throw new Error("Method hasn't been implemented yet!");
+  const hashMap = {};
+
+  for(string of strings) {
+    const sorted = string
+    .split('')
+    .sort()
+    .join('')
+
+    hashMap[sorted] ? hashMap[sorted].push(string) : hashMap[sorted] = [string];
+  }
+
+  return Object.values(hashMap);
 }
 
 // This method will return the k most common elements
 // in the case of a tie it will select the first occuring element.
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: o(n)
+// Space Complexity: o(n)
 function top_k_frequent_elements(list, k) {
-  throw new Error("Method hasn't been implemented yet!");
-}
+  let hash = {}
+  kArr = []
 
+  if (list.length === 0) return [];
+
+  list.forEach(element => {
+    hash[element] ? hash[element] += 1 : hash[element] = 1;
+  })
+
+  for (let i=0; i<=list.length; i++) {
+    if (hash[i] >= k) {
+      kArr.push(i);
+    }
+  }
+
+  if (kArr.length === 0) return list;
+  if (kArr.length === list.length) return [list[0]];
+  
+  return kArr;
+}
 
 // This method will return true if the table is still
 //   a valid sudoku table.
 //   Each element can either be a ".", or a digit 1-9
 //   The same digit cannot appear twice or more in the same 
 //   row, column or 3x3 subgrid
-// Time Complexity: ?
-// Space Complexity: ?
+// Time Complexity: o(1)
+// Space Complexity: o(n)
+// the time is dependent on the the rate of growth which in this case is const
 function valid_sudoku(table) {
```

Comment:
👍 This is really compact.  Well done.  Since a Sudoku board never increases in size, this is O(1) for both time/space complexity.